File: 25. Callback function demo.cpp
// Topic: Callback function - pass function pointer as argument
#include <iostream>
using namespace std;

void caller(void (*cb)(int))
{
    cb(42);
}

void myCallback(int x)
{
    cout << "Callback called with " << x << '\n';
}

int main()
{
    caller(myCallback);
    return 0;
}

// Time Complexity: O(1) for callback invocation.
// Space Complexity: O(1).

File: 25. Callback function demo.py
"""
Callback demo: pass function as argument
"""
def caller(cb):
    cb(42)

def my_callback(x):
    print('Callback called with', x)

def main():
    caller(my_callback)

if __name__ == '__main__':
    main()

# Time Complexity: O(1).
# Space Complexity: O(1).
